### DISCUSSION TITLE
An unexpected error has occurred.

### DISCUSSION THREAD

**Discussion by ericksonholguin** [vercel/next.js#19904] [Help] [answered]
I finally finished creating my project made with nestjs, but when I try to upload it, it shows me the following error `An unexpected error has occurred.`
![image](https://user-images.githubusercontent.com/38663299/101286887-838ae380-37c3-11eb-937f-ecfeeeb2879a.png)

As long as I enter with the production url, since when I enter with the url that generates the deploy, it lets me access it on some occasions

this is what the chrome console throws me

![image](https://user-images.githubusercontent.com/38663299/101286954-eda38880-37c3-11eb-86bf-18aebdc87305.png)

**Comment by jamesmosier** [comment, marked as answer]
Digging into the minified source code, it looks like the issue is somewhere in your application near the text `No hay publicaciones recientes` and the issue is that you are doing a check for `.length` on a variable that is undefined. Therefore to debug you could first determine where you have a `.length` near that text and then figure out why the variable is undefined. It is likely that some data is not being fetched.

<img width="560" alt="Screen Shot 2020-12-06 at 8 58 58 PM" src="https://user-images.githubusercontent.com/2854919/101301146-e3a27980-3805-11eb-92a5-11abc09b7eca.png">

---

It is caused because at some point `recentsPosts` becomes `undefined`. So then when you check for `.length` on that it throws an error.

<img width="353" alt="Screen Shot 2020-12-06 at 9 05 55 PM" src="https://user-images.githubusercontent.com/2854919/101301448-dafe7300-3806-11eb-8619-18f4a7d2adbc.png">
